From 3290639e8de271a68991060097b8e18012aac306 Mon Sep 17 00:00:00 2001
From: Adrian-Ken Rueegsegger <ken@codelabs.ch>
Date: Tue, 3 Oct 2017 16:44:51 +0200
Subject: [PATCH 1/3] mutime: Make Sinfo an imported constant

This fixes a warning issued by gnatprove regarding the initialization of
Sinfo. Also adjust refinement of abstract timer state to match global
contract of Hz function.
---
 ada/mutime/hw-time-timer.adb | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/ada/mutime/hw-time-timer.adb b/ada/mutime/hw-time-timer.adb
index 272289c..e7d80e1 100644
--- a/ada/mutime/hw-time-timer.adb
+++ b/ada/mutime/hw-time-timer.adb
@@ -17,16 +17,17 @@ with Musinfo;
 with Muschedinfo;
 
 package body HW.Time.Timer
-   with Refined_State => (Timer_State => null,
-                          Abstract_Time => (Sinfo, Sched_Info))
+   with Refined_State => (Timer_State => Sinfo,
+                          Abstract_Time => Sched_Info)
 is
    Sinfo_Base_Address : constant := 16#000e_0000_0000#;
    Sinfo_Page_Size    : constant
      := ((Musinfo.Subject_Info_Type_Size + (16#1000# - 1))
          / 16#1000#) * 16#1000#;
 
-   Sinfo : Musinfo.Subject_Info_Type
+   Sinfo : constant Musinfo.Subject_Info_Type
    with
+      Import,
       Address => System'To_Address (Sinfo_Base_Address);
 
    Sched_Info : Muschedinfo.Scheduling_Info_Type
-- 
2.1.4

